app/models/cloud_tenant.rb

```diff
@@ -8,7 +8,6 @@ class CloudTenant < ApplicationRecord
   has_many   :cloud_subnets
   has_many   :network_ports
   has_many   :network_routers
-  has_many   :shared_cloud_networks, -> { where :shared => true }, :through => :ext_management_system, :source => :cloud_networks
```

Comment:
I think the provision workflow can just have

src_obj.try(:all_cloud_netowrks) || scr_obj.cloud_networks

And I will define all cloud networks just on cloud_tenant.